pkg/webhook/admission/defaulter_handler.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admission
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// DefaulterFor defines functions for setting defaults on resources.
+type DefaulterFor interface {
+	For
+	Default(ctx context.Context, obj runtime.Object)
```

Comment:
```suggestion
	Default(ctx context.Context, obj runtime.Object) error
```

Reply:
Why?

Reply:
This could be handy if for any reason the default logic could fail (e.g we make an API call inside the defaulter)

Reply:
Fair enough :), added